src/snowflake/snowpark/modin/plugin/extensions/series_overrides.py

```diff
@@ -392,6 +392,25 @@ def __init__(
         self.name = name
 
 
+@register_series_accessor("_update_inplace")
+def _update_inplace(self, new_query_compiler) -> None:
+    """
+    Update the current Series in-place using `new_query_compiler`.
+
+    Parameters
+    ----------
+    new_query_compiler : BaseQueryCompiler
+        QueryCompiler to use to manage the data.
+    """
+    super(Series, self)._update_inplace(new_query_compiler=new_query_compiler)
+    # Propagate changes back to parent so that column in dataframe had the same contents
+    if self._parent is not None:
+        if self._parent_axis == 1 and isinstance(self._parent, DataFrame):
```

Comment:
Can the parent ever be an Index? I don't think so, but just checking

Reply:
Good question - I don't think so - I checked `to_series` and it doesn't seem to set the parent:
```python
In [1]: im();
a =
In [2]: a = pd.Index([1, 2, 3, 4])

In [3]: a.to_series()
Out[3]: a
1    1
2    2
3    3
4    4
dtype: int64

In [4]: a.to_series()._parent
```
Additionally, Index's have no `iloc`, and indexing returns another Index object, so I don't think it should be possible?
```
In [1]: im();
a =
In [2]: a = pd.Index([1, 2, 3, 4])

In [3]: type(a.iloc[:2])
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Cell In[3], line 1
----> 1 type(a.iloc[:2])

File ~/Documents/snowpark-python/src/snowflake/snowpark/modin/plugin/extensions/index.py:282, in Index.__getattr__(self, key)
    278     if hasattr(native_index, key):
    279         # Any methods that not supported by the current Index.py but exist in a
    280         # native pandas index object should raise a not implemented error for now.
    281         ErrorMessage.not_implemented(f"Index.{key} is not yet implemented")
--> 282 raise err

File ~/Documents/snowpark-python/src/snowflake/snowpark/modin/plugin/extensions/index.py:274, in Index.__getattr__(self, key)
    256 """
    257 Return item identified by `key`.
    258
   (...)
    271 of current scope that are not implemented.
    272 """
    273 try:
--> 274     return object.__getattribute__(self, key)
    275 except AttributeError as err:
    276     if not key.startswith("_"):

AttributeError: 'Index' object has no attribute 'iloc'

In [4]: type(a[:2])
Out[4]: snowflake.snowpark.modin.plugin.extensions.index.Index
```